network/src/connectivity_manager/mod.rs

```diff
@@ -347,10 +347,12 @@ where
         // It enforces that a full node cannot have more outgoing connections than `connection_limit`
         // including in flight dials.
         let to_connect_size = if let Some(conn_limit) = self.connection_limit {
-            min(
-                conn_limit - self.connected.len() - self.dial_queue.len(),
-                to_connect.len(),
-            )
+            let current = self.connected.len() + self.dial_queue.len();
+            if current < conn_limit {
+                min(conn_limit - current, to_connect.len())
+            } else {
+                0
+            }
```

Comment:
https://doc.rust-lang.org/std/primitive.u32.html
this can be written as
```
conn_limit.checked_sub(self.connected.len()).and_then(|v| v.checked_sub(to_connect.len())).unwrap_or(0);
```
or you can do the add too

Reply:
`saturating_sub` does `checked_sub().unwrap_or(0)`